src/include/stats/abstract_metric.h

```diff
@@ -0,0 +1,310 @@
+#pragma once
+
+#include <emmintrin.h>
+#include <atomic>
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "catalog/catalog_defs.h"
+#include "stats/abstract_raw_data.h"
+#include "stats/statistic_defs.h"
+
+namespace terrier {
+
+namespace transaction {
+class TransactionContext;
+}  // namespace transaction
+
```

Comment:
This is a forward declaration for methods of AbstractMetric, which have TransactionContext * as parameters

Reply:
Since you are using `transaction::TransactionContext *` as parameter types, I think this is not necessary. I tried compiling the code without this declaration and it compiled successfully